compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -1073,7 +1064,9 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
             todo.extend(static_roots(self));
             while let Some(id) = todo.pop() {
```

Comment:
I would have loved to also port this recursive traversal over to `visit_reachable_allocs`... but `visit_reachable_allocs` calls `get_alloc_raw` which can fail, and this here is in the context of a function that cannot fail, so it didn't seem worth it unfortunately.